crates/api/src/fetch.rs

```diff
@@ -0,0 +1,28 @@
+//! Kitsune2 fetch queue types.
+
+use std::sync::Arc;
+
+use crate::{AgentId, OpId};
+
+/// Trait for implementing a fetch queue for managing ops
+/// to be fetched from other agents.
+pub trait FetchQueueT: 'static + Send + Sync {
+    /// Add op ids to be fetched to the queue.
+    fn add_ops(&mut self, op_list: Vec<OpId>, source: AgentId);
+
+    /// Get a list of op ids to be fetched.
+    fn get_ops_to_fetch(&self) -> Vec<OpId>;
+
+    /// Get a random agent that holds the ops to be fetched.
+    fn get_random_source(&self) -> Option<AgentId>;
+}
+
+/// Trait-object fetch queue instance.
+pub type DynFetchQueue = Arc<dyn FetchQueueT>;
+
+/// Trait for implementing a fetch task that periodically attempts to
+/// send op fetch requests to another agent.
+pub trait FetchTaskT {
+    /// Spawn a new fetch task, backed by a fetch queue.
+    fn spawn(&self, fetch_queue: DynFetchQueue);
+}
```

Comment:
I don't think we need the complexity of a separate "task". Following the abstraction set out in the spike, we need a FetchQueueFactory from which the space module in *its* factory can construct a real `DynFetchQueue` instance. That instance internally may have tasks it may have data stores... as far as abstractions go, we don't care, that's up to the implementation.